File: LeetCode root folder/Greatest Common Divisor of Strings.java
// 9%


class Solution {
    public String gcdOfStrings(String str1, String str2) {
        int l1 = str1.length(), l2 = str2.length();
        String res = "";
        for (int i = 1; i <= Math.max(l1, l2); i++) {
            if (l1 % i == 0 && l2 % i == 0 && str1.substring(0, i).equals(str2.substring(0, i))) {
                boolean flag1 = true, flag2 = true;
                for (int j = i; j <= l1-i; j+=i) {
                    if (!str1.substring(j-i, j).equals(str1.substring(j, j+i))) {
                        flag1 = false;
                        break;
                    }
                }
                for (int j = i; j <= l2-i; j+=i) {
                    if (!str2.substring(j-i, j).equals(str2.substring(j, j+i))) {
                        flag2 = false;
                        break;
                    }
                }
                if (flag1 && flag2) {
                    res = str1.substring(0, i);
                }
            }
        }
        return res;
    }
}

File: LeetCode root folder/Greatest Common Divisor of Strings.py
# 25%
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        l1, l2 = len(str1), len(str2)
        res = ""
        for i in range(1, max(l1, l2)+1):
            if l1 % i == 0 and l2 % i == 0 and str1[:i] == str2[:i]:

                flag1, flag2 = False, False
                for j in range(i, l1-i+1, i):
                    if str1[j-i:j] != str1[j:j+i]:
                        break
                else:
                    flag1 = True
                for j in range(i, l2-i+1, i):
                    if str2[j-i:j] != str2[j:j+i]:
                        break
                else:
                    flag2 = True

                if flag1 and flag2:
                    res = str1[:i]
        return res 